helix-core/src/indent.rs

```diff
@@ -293,23 +271,20 @@ pub fn suggested_indent_for_pos(
     syntax: Option<&Syntax>,
     text: RopeSlice,
     pos: usize,
+    line: usize,
     new_line: bool,
-) -> usize {
+) -> Option<usize> {
     if let (Some(query), Some(syntax)) = (
         language_config.and_then(|config| config.indent_query()),
         syntax,
     ) {
         let byte_start = text.char_to_byte(pos);
         let node = get_highest_syntax_node_at_bytepos(syntax, byte_start);
-
-        // let config = load indentation query config from Syntax(should contain language_config)
-
         // TODO: special case for comments
         // TODO: if preserve_leading_whitespace
-        calculate_indentation(query, node, new_line)
+        Some(calculate_indentation(query, node, line, new_line))
     } else {
-        // TODO: heuristics for non-tree sitter grammars
-        0
+        None
```

Comment:
I was thinking you could just call `indent_level_for_line` here? Hence the TODO

Reply:
In the current version we can't really do it here, because the indentation in this case also depends on the (previous) cursor position: The indentation of the new line is the indentation of the current line (which could be above or below depending on whether you us `o` or `O`). Because it doesn't (just) depend on the position of the newline, including it in `suggested_indent_for_pos` doesn't really make sense.

I'm not sure if this indentation behavior is what we want, we could also simply copy the indentation from the line above. Then we could also include it in `suggested_indent_for_pos` and the behavior would be the same for `o` and `Enter` in insert mode. I'm not super happy with the current solution either so let me know if you have any other ideas.